src/requestLib.js

```diff
@@ -1,44 +1,47 @@
 var http = require('http');
+var https = require('https');
 var constants = require('../config/constants');
 var Utils = require('./utils');
-var keepAliveAgent = new http.Agent({
-  keepAlive: true
-});
+var url = require('url');
+var HttpProxyAgent = require('https-proxy-agent');
+var HttpsProxyAgent = require('https-proxy-agent');
+
 var RdGlobalConfig = constants.RdGlobalConfig;
 
 var RequestLib = {
   /**
    * Method to perform the request on behalf of the client
+   * @param {schemeObj: Object} schemeObj 
    * @param {{request: Object, furtherRequestOptions: Object}} params 
    * @param {http.IncomingMessage} clientRequest 
    * @param {Number} retries 
    */
-  _makeRequest: function (params, clientRequest, retries) {
+  _makeRequest: function (schemeObj, params, clientRequest, retries) {
     return new Promise(function (resolve, reject) {
-      var requestOptions = Object.assign({}, params.furtherRequestOptions, {
-        agent: keepAliveAgent
-      });
-
-      // Adding a custom header for usage and debugging purpose at BrowserStack
-      requestOptions.headers['X-Requests-Debugger'] = clientRequest.id;
-
-      // Initialize the request to be fired on behalf of the client
-      var request = http.request(requestOptions, function (response) {
+      var requestOptions = Object.assign({}, params.furtherRequestOptions);
+      requestOptions.agent = new schemeObj.Agent({keepAlive: true});
+      if(RdGlobalConfig.proxy) { 
+        var proxyOpts = url.parse(`${RdGlobalConfig.proxy.host}:${RdGlobalConfig.proxy.port}`);
+        if(RdGlobalConfig.proxy.username && RdGlobalConfig.proxy.password);
+          proxyOpts.auth = `${RdGlobalConfig.proxy.username}:${RdGlobalConfig.proxy.password}`;
+        requestOptions.agent = RdGlobalConfig.SCHEME == 'http' ? new HttpProxyAgent(proxyOpts) : new HttpsProxyAgent(proxyOpts);
+      }
```

Comment:
This can be taken out. Creating the agent objects for every new request is not required since we would be aware from the beginning itself whether we have an upstream proxy or not. Similarly for the keep-alive agent.

Reply:
Done

Reply:
can be resolved.